src/main/java/com/teradata/tpcds/ScalingInfo.java

```diff
@@ -69,21 +69,41 @@ public long getRowCountForScale(int scale)
         }
     }
 
-    private long computeCountUsingStaticScale()
+    private long computeCountUsingStaticScale(float scale)
     {
-        return getRowCountForScale(1);
+        if (scale == Math.round(scale)) {
+            return getRowCountForScale(1);
+        }
+        // Handle decimal scales (not part of spec). Will only have an effect on
+        // decimal scales 0 < x < 1.
```

Comment:
I did it because otherwise we don't get the required scaling and we're way off in terms of output size. I thought that as long as we're breaking the standard why not break away from it fully.